### DISCUSSION TITLE
`toFirestore` inside `globalFirestoreOptions.converter` is never called

### DISCUSSION THREAD

**Discussion by mrleblanc101** [vuejs/vuefire#1499] [Questions and Help] [answered]
### Reproduction

.

### Steps to reproduce the bug

I created this global converter, the fromFirestore gets called, but not the toFirestore.
I see `test 2` in the console, but never `test 1`, and the data in my firestore does not have the field `created_at`.
```js
globalFirestoreOptions.converter = {
    toFirestore(docData: any) {
        console.log('test 1');
        return {
            created_at: serverTimestamp(),
            ...docData,
        }
    },
    fromFirestore(snapshot: QueryDocumentSnapshot, options: SnapshotOptions): DocumentData {
        console.log('test 2');
        const data = firestoreDefaultConverter.fromFirestore(snapshot)!;
        return data;
    },
};
```

### Expected behavior

Both toFirestore and fromFirestore should be called.

### Actual behavior

Only the fromFirestore is called.

### Additional information

I want to automatically add a `created_at` field so that I can sort by date when fetching.
> Important: Unlike "push IDs" in the Firebase Realtime Database, Cloud Firestore auto-generated IDs do not provide any automatic ordering. If you want to be able to order your documents by creation date, you should store a timestamp as a field in the documents.
Ref: https://firebase.google.com/docs/firestore/manage-data/add-data?hl=fr

**Comment by mrleblanc101** [comment, marked as answer]
It works if I use `.withConverter` instead.
Exemple:
```js
const addDefaultFields = {
    toFirestore(docData: DocumentData): DocumentData {
        console.log('test 1');
        return {
            created_at: serverTimestamp(),
            ...docData,
        }
    },
    fromFirestore(snapshot: QueryDocumentSnapshot, options: SnapshotOptions): DocumentData {
        console.log('test 2');
        const data = firestoreDefaultConverter.fromFirestore(snapshot)!;
        return data;
    },
};

...

async function addProject(option: Project) {
    const project = await addDoc(collection(db, 'projects').withConverter(addDefaultFields), {
        name: option.name,
        user: user.value!.uid,
    })
    return { id: project.id, name: option.name };
}
```

**Reply by posva** [reply]
You need to do `.withConverter()`. VueFire cannot add that for you. It only applies to `useDocument()` and others from vuefire

**Reply by mrleblanc101** [reply]
I'm not quite sure I understand, I thought `globalFirestoreOptions.converter` was a way to define a global converter.

**Reply by mrleblanc101** [reply]
Oh I think it get it, `addDoc` is from firebase directly while `useDocument` is from vue-fire. So `globalFirestoreOptions.converter` only affect vue-fire call and not firebase call

**Reply by mrleblanc101** [reply]
So if I understand correctly the `globalFirestoreOptions.converter.toFirebase` is never called since you can't push data using vue-fire as you need to use the firebase `addDoc, updateDoc` function directly 🤔
